schema/string.go

```diff
@@ -17,18 +17,24 @@ const (
 	stringUUIDVersion = 4
 )
 
-func checkStringConstraints(v string, minLength, maxLength int, t string) error {
+func checkStringConstraints(v string, minLength, maxLength int, pattern *regexp.Regexp) error {
 	if minLength != 0 && len(v) < minLength {
-		return fmt.Errorf("constraint check error: %s:%v %v < minimum:%v", t, v, minLength)
+		return fmt.Errorf("constraint check error: %v %v < minimum:%v", v, len(v), minLength)
 	}
 	if maxLength != 0 && len(v) > maxLength {
-		return fmt.Errorf("constraint check error: %s:%v %v > maximum:%v", t, v, maxLength)
+		return fmt.Errorf("constraint check error: %v %v > maximum:%v", v, len(v), maxLength)
+	}
+	if pattern != nil {
+		match := pattern.MatchString(v)
+		if false == match {
+			return fmt.Errorf("constraint check error: %v don't fit pattern : %v ", v, pattern)
```

Comment:
I am not sure what is the result if you print the compiled pattern. The best here is to print the string version of the pattern. Another reason to pass the whole Constraints here.

Reply:
Actually it will show its pattern,  but I agree with you, change it to pattern(string) is better.

Reply:
I think we should print out the compiled pattern to show the pattern instead of the Constraints.Pattern, because we won't pass any pattern(string) to Constraints in our testing function (string_test.go).